packages/fxa-shared/db/redis.ts

```diff
@@ -188,83 +192,138 @@ export class RedisShared {
   }
 
   async del(key: string) {
-    return await this.redis.del(key);
+    const result = await this.redis.del(key);
+    return result;
   }
 
   async getRefreshTokens(uid: Buffer | string) {
+    this.metrics?.increment('redis.getRefreshTokens');
+    const span = tracer.startSpan('redis.getRefreshTokens');
     try {
       const p1 = this.redis.hgetall(hex(uid));
       const p2 = this.resolveInMs(p1, this.timeoutMs, {});
       const tokens = await Promise.race([p1, p2]);
+      span.setAttribute(
+        'redis.getRefreshTokens.tokens.length',
+        Object.keys(tokens).length
+      );
+      this.metrics?.histogram(
+        'redis.getRefreshTokens.tokens.length',
+        Object.keys(tokens).length
+      );
       for (const id of Object.keys(tokens)) {
         tokens[id] = RefreshTokenMetadata.parse(tokens[id]);
       }
       return tokens;
     } catch (e) {
+      this.metrics?.increment('redis.getRefreshTokens.error');
       this.log?.error('RedisShared', { error: e });
       return {};
+    } finally {
+      span.end();
     }
   }
 
   async pruneSessionTokens(uid: string, tokenIds: string[] = []) {
+    this.metrics?.increment('redis.pruneSessionTokens');
+    const span = tracer.startSpan('redis.pruneSessionTokens');
     const p1 = this.redis.pruneSessionTokens(uid, JSON.stringify(tokenIds));
     const p2 = this.rejectInMs(p1, this.timeoutMs);
-    return Promise.race([p1, p2]);
+    const result = Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
   async pruneRefreshTokens(
     uid: Buffer | String,
     tokenIdsToPrune: Buffer[] | string[]
   ) {
+    this.metrics?.increment('redis.pruneRefreshTokens');
+    const span = tracer.startSpan('redis.pruneRefreshTokens');
     const p1 = this.redis.hdel(hex(uid), ...tokenIdsToPrune.map((v) => hex(v)));
     const p2 = this.resolveInMs(p1, this.timeoutMs);
-    return await Promise.race([p1, p2]);
+    const result = await Promise.race([p1, p2]);
```

Comment:
nit: Might be nothing, but noticed that the `Promise.race([])` above on L232 doesn't have an `await`, but this line does.
 In fact, the non-`await`ed version might be the outlier.

if I Ctrl+F ` = await Promise.race(` [in this PR], I get 9 results.
If I Ctrl+F ` = Promise.race(`, I get 1 result.

Reply:
Yes, this is an existing inconsistency in the code. Technically, it doesn't matter as long as the code returns a promise; however, I don't like inconsistency, so I'll try to keep the lines I changed consistent at least.